controller/raft/mesh/timeouts.go

```diff
@@ -0,0 +1,42 @@
+package mesh
+
+import (
+	"sync"
+	"time"
+)
+
+func newDeadline() *deadline {
+	return &deadline{
+		C: make(chan struct{}, 1),
+	}
+}
+
+type deadline struct {
```

Comment:
yeah, I thought I'd see if I either figured out a better way to do it, or if it was used in more places. Then it would either go away, or get moved to a common place.